### PR TITLE
perf(Trace): use faster way to copy one array into another

### PR DESCRIPTION
**Pull request by kibertoad**
## Description

This is a follow-up to https://github.com/newrelic/node-newrelic/pull/1774, optimizing the solution further. This is the fastest way to copy values from one array into another array, see https://github.com/kibertoad/nodejs-benchmark-tournament/blob/master/add-array/_results/results.md for benchmarking results (pushDestruct method, while currently `pushForEach` is used, which is the slowest way to do so)

This retains the same logic as is currently used, but works faster (rougly 25% speed-up for the copying operation).

## How to Test

I've benchmarked the algorithm to ensure that this is the fastest possible option. See https://github.com/kibertoad/nodejs-benchmark-tournament/blob/master/add-array/contestants/pushDestruct.js for how this was implemented in benchmark.

## Related Issues

https://github.com/newrelic/node-newrelic/pull/1774

### CODE REVIEW

**Comment by kibertoad** [comment]
@bizob2828 ping

**Review by bizob2828 — APPROVED** [review]

**Comment by bizob2828** [comment]
@kibertoad you'll have to rebase.  There was an unrelated test fix for pino 8.15.1. that I just landed in #1782

**Comment by kibertoad** [comment]
@bizob2828 It's not possible to rebase on merge?

File: lib/transaction/trace/index.js [hunk 1]
```diff
@@ -224,7 +224,7 @@ Trace.prototype.getTotalTimeDurationInMillis = function getTotalTimeDurationInMi
   while (segments.length !== 0) {
     const segment = segments.pop()
     totalTimeInMillis += segment.getExclusiveDurationInMillis()
-    segment.getChildren().forEach((childSegment) => segments.push(childSegment))
+    segments.push(...segment.getChildren())
```
Comment (bizob2828) [inline lib/transaction/trace/index.js:227]: It looks like we have a test that creates a lot of child segments and then contains stacks.  This test is now failing by spreading the children and pushing vs iterating and pushing

**Review by bizob2828 — CHANGES_REQUESTED** [review]
Although this is faster it appears there are historical reason why it is iterating and pushing

**Comment by kibertoad** [comment]
Right, so this PR should be adjusted to just use the for loop?

**Comment by kibertoad** [comment]
@bizob2828 Yeah, that was my thought. Should we do that, or should we just close this PR?

**Comment by bizob2828** [comment]
> @bizob2828 Yeah, that was my thought. Should we do that, or should we just close this PR?

It's up to you. Honestly, the initial PR was the first time I ever looked at this code.  It's very old and thankfully we have some integration tests to expose the nuance that you hit.

**Comment by bizob2828** [comment]
I'm going to close this since there's been 0 activity for a month. Feel free to reopen with my suggestions addressed.

**Pull request closed without merging**